tests/common/test_cgroupapi.py

```diff
@@ -317,3 +317,90 @@ def mock_popen(*args, **kwargs):
 
             self.assertTrue(cpu_found, 'start_extension_command did not return a cpu cgroup')
             self.assertTrue(memory_found, 'start_extension_command did not return a memory cgroup')
+
+    @skip_if_predicate_false(i_am_root, "Test does not run when normal user")
+    def test_start_extension_command_should_use_systemd_if_not_failing(self):
+        original_popen = subprocess.Popen
+
+        def mock_popen(*args, **kwargs):
+            return original_popen(*args, **kwargs)
+
+        with tempfile.TemporaryFile(dir=self.tmp_dir, mode="w+b") as stdout:
+            with tempfile.TemporaryFile(dir=self.tmp_dir, mode="w+b") as stderr:
+                with patch("azurelinuxagent.common.cgroupapi.subprocess.Popen", side_effect=mock_popen) as patch_mock_popen:
+                    process, extension_cgroups = SystemdCgroupsApi().start_extension_command(
+                        extension_name="Microsoft.Compute.TestExtension-1.2.3",
+                        command="date",
+                        shell=True,
+                        cwd=self.tmp_dir,
+                        env={},
+                        stdout=stdout,
+                        stderr=stderr)
+
+                    process.wait()
+
+                    ret = process.poll()
+
+                    # We should have invoked the extension command only once and succeeded
+                    self.assertEquals(0, ret)
+                    self.assertEquals(1, patch_mock_popen.call_count)
```

Comment:
You should also assert that the command executed by popen includes "systemd-run"